crabs/detection_tracking/detection_utils.py

```diff
@@ -226,65 +229,243 @@ def calculate_iou(box1, box2) -> float:
+    """
+    ground_truth_data = []
+    max_frame_number = 0
+
+    # Open the CSV file and read its contents line by line
+    with open(gt_dir, "r") as csvfile:
+        csvreader = csv.reader(csvfile)
+        next(csvreader)  # Skip the header row
+        for row in csvreader:
+            # Extract relevant information from each row
+            filename = row[0]
+            region_shape_attributes = json.loads(row[5])
+            region_attributes = json.loads(row[6])
+
+            # Extract bounding box coordinates and object ID
+            x = region_shape_attributes["x"]
+            y = region_shape_attributes["y"]
+            width = region_shape_attributes["width"]
+            height = region_shape_attributes["height"]
+            track_id = region_attributes["track"]
+
+            # Compute the frame number from the filename
+            frame_number = int(filename.split("_")[-1].split(".")[0])
+            frame_number = frame_number - 1
+
+            # Update max_frame_number
+            max_frame_number = max(max_frame_number, frame_number)
+
+            # Append the extracted data to the list
+            ground_truth_data.append(
+                {
+                    "frame_number": frame_number,
+                    "x": x,
+                    "y": y,
+                    "width": width,
+                    "height": height,
+                    "id": track_id,
+                }
+            )
+
+    # Initialize a list to store the ground truth bounding boxes for each frame
+    gt_boxes_list = [np.array([]) for _ in range(max_frame_number + 1)]
+
+    # Organize ground truth data into gt_boxes_list
+    for data in ground_truth_data:
+        frame_number = data["frame_number"]
+        bbox = np.array(
+            [
+                data["x"],
+                data["y"],
+                data["x"] + data["width"],
+                data["y"] + data["height"],
+                data["id"],
+            ],
+            dtype=np.float32,
+        )
+        gt_boxes_list[frame_number] = (
+            np.vstack([gt_boxes_list[frame_number], bbox])
+            if gt_boxes_list[frame_number].size
+            else bbox
+        )
+    return gt_boxes_list
+
+
+def write_bbox_to_csv(
+    bbox: np.ndarray,
+    frame: np.ndarray,
+    frame_name: str,
+    csv_writer: Any,
 ) -> None:
     """
-    Plot ground truth bounding boxes.
+    Write bounding box annotation to a CSV file.
 
-    Parameters
-    ----------
-    ground_truths : dict
-        A dictionary mapping object IDs to their corresponding ground truth annotations.
-    frame_number : int
-        The frame number to track.
-    tracked_boxes : np.ndarray
-        An array containing sorted bounding boxes of detected objects.
-    iou_threshold : float
-        The intersection over union threshold for considering a match.
-    frame_copy : np.ndarray
-        A copy of the input frame for drawing bounding boxes.
+    Parameters:
+    -----------
```

Comment:
```suggestion
    Parameters
    ----------
```
Bike-sheddyest of all of the bikeshedding comments: [numpy doc style doesn't need `:` if you're underlining](https://numpydoc.readthedocs.io/en/latest/format.html#parameters).

Reply:
Be nice if we could add a linter to just solve all of this for us. A quick search found [Velin](https://github.com/Carreau/velin) which claims to work with pre-commit. Worth a try? (Obvs. not in this PR)

Reply:
Sorry my bad. I used to use different style, and have a lot typo while porting.